src/System.Text.Json/tests/PerfSmokeTests.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text.Json.Tests.Resources;
+using Xunit;
+
+[assembly: CollectionBehavior(DisableTestParallelization = true)]
+
+namespace System.Text.Json.Tests
+{
+    public class PerfSmokeTests
+    {
+        private static readonly Stopwatch Timer = new Stopwatch();
+        private const bool Collect = true;
+        private const int MemoryToleranceFactor = 5;
+
+        // Keep < 10. Otherwise tests take too long to execute (over 10 seconds).
+        private const int NumberOfIterations = 5;
+
+        private const int NumberOfSamples = 3;
+        private const int ExpectedMemoryBenchMark = 3000000;
+
+        private static readonly bool OutputResults = false;
+
+        [Fact]
+        public void ReadBasicJson()
+        {
+            Output("====== TEST ReadBasicJson ======");
+            RunTest(TestJson.BasicJson, 100 + NumberOfIterations, ExpectedMemoryBenchMark*MemoryToleranceFactor);
```

Comment:
When are the resources read? I think you need to ensure that everything is loaded into memory (byte[]) before starting the times. Otherwise the test might have lots of noise associated with resource access.